src/SourceLocator/Exception/InvalidDirectory.php

```diff
@@ -0,0 +1,7 @@
+<?php
+
+namespace BetterReflection\SourceLocator\Exception;
+
+class InvalidDirectory extends  \RuntimeException
+{
```

Comment:
Might be beneficial to have two named constructors: one for non-string values, one for a string value (which can also tell us if the given path exists, and why it is not recognized as directory)